src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_pe/PlayerListSetEntry.java

```diff
@@ -56,21 +57,25 @@ public PlayerListSetEntry(ConnectionImpl connection) {
 				PESkinsProvider skinprovider = PESkinsProviderSPI.getProvider();
 				ClientBoundPacketData serializer = ClientBoundPacketData.create(PEPacketIDs.PLAYER_INFO);
 				serializer.writeByte(0);
-				VarNumberSerializer.writeVarInt(serializer, infos.size());
+				VarNumberSerializer.writeVarInt(serializer, infos.size() - 1);
```

Comment:
Woah.. What if the list is send by plugin etc, and contains not the player itself?
To fix maybe employ fixed varint write and store writerindex, then overwrite with fixed varint with correct length.

Reply:
Kind of like the implementation in EntityMetadata middle packet.